data/sea/54-zebra.h

```diff
@@ -87,25 +112,38 @@ static ierror_msg_t zebra_translate
         fact_count++;
     }
 
+    /* the number of unread facts for this entity */
+    iint_t fact_offset    = state->entity_fact_offset;
+    iint_t fact_remaining = fact_count == 0 ? 0 : fact_count - fact_offset;
```

Comment:
yeah, this was wrong -- `fact_count == 0` should only be when we have not read any fact for this entity. That should be an invariant actually.